src/Communication/NetworkApi.php

```diff
@@ -0,0 +1,264 @@
+use JaxkDev\DiscordBot\Models\Role;
+use JaxkDev\DiscordBot\Models\User;
+use JaxkDev\DiscordBot\Models\VoiceState;
+use JaxkDev\DiscordBot\Models\Webhook;
+
+final class NetworkApi{
+
+    //Version will change for any breaking changes to:
+    //Models, Packets, Magic.
+    public const VERSION = 1;
+    public const MAGIC = 0x4a61786b; //Jaxk (max 4 bytes)
+
+    /**
+     * @var array<int, class-string<Packet>>
+     */
+    public const PACKETS_MAP = [
+        //1-2 Misc Packets
+        Heartbeat::SERIALIZE_ID => Heartbeat::class,
+        Resolution::SERIALIZE_ID => Resolution::class,
+
+        //3-4 External->PMMP Packets
+        Connect::SERIALIZE_ID => Connect::class,
+        Disconnect::SERIALIZE_ID => Disconnect::class,
+
+        //5-38 Discord->PMMP Packets
+        DiscordConnected::SERIALIZE_ID => DiscordConnected::class,
+        BanCreate::SERIALIZE_ID => BanCreate::class,
+        BanDelete::SERIALIZE_ID => BanDelete::class,
+        ChannelCreate::SERIALIZE_ID => ChannelCreate::class,
+        ChannelDelete::SERIALIZE_ID => ChannelDelete::class,
+        ChannelPinsUpdate::SERIALIZE_ID => ChannelPinsUpdate::class,
+        ChannelUpdate::SERIALIZE_ID => ChannelUpdate::class,
+        GuildJoin::SERIALIZE_ID => GuildJoin::class,
+        GuildLeave::SERIALIZE_ID => GuildLeave::class,
+        GuildUpdate::SERIALIZE_ID => GuildUpdate::class,
+        InviteCreate::SERIALIZE_ID => InviteCreate::class,
+        InviteDelete::SERIALIZE_ID => InviteDelete::class,
+        MemberJoin::SERIALIZE_ID => MemberJoin::class,
+        MemberLeave::SERIALIZE_ID => MemberLeave::class,
+        MemberUpdate::SERIALIZE_ID => MemberUpdate::class,
+        MessageDelete::SERIALIZE_ID => MessageDelete::class,
+        MessageReactionAdd::SERIALIZE_ID => MessageReactionAdd::class,
+        MessageReactionRemove::SERIALIZE_ID => MessageReactionRemove::class,
+        MessageReactionRemoveAll::SERIALIZE_ID => MessageReactionRemoveAll::class,
+        MessageReactionRemoveEmoji::SERIALIZE_ID => MessageReactionRemoveEmoji::class,
+        MessageSent::SERIALIZE_ID => MessageSent::class,
+        MessageUpdate::SERIALIZE_ID => MessageUpdate::class,
+        PresenceUpdate::SERIALIZE_ID => PresenceUpdate::class,
+        RoleCreate::SERIALIZE_ID => RoleCreate::class,
+        RoleDelete::SERIALIZE_ID => RoleDelete::class,
+        RoleUpdate::SERIALIZE_ID => RoleUpdate::class,
+        VoiceStateUpdate::SERIALIZE_ID => VoiceStateUpdate::class,
+        InteractionReceived::SERIALIZE_ID => InteractionReceived::class,
+        MessageDeleteBulk::SERIALIZE_ID => MessageDeleteBulk::class,
+        BotUserUpdate::SERIALIZE_ID => BotUserUpdate::class,
+        WebhooksUpdate::SERIALIZE_ID => WebhooksUpdate::class,
+        ThreadCreate::SERIALIZE_ID => ThreadCreate::class,
+        ThreadDelete::SERIALIZE_ID => ThreadDelete::class,
+        ThreadUpdate::SERIALIZE_ID => ThreadUpdate::class,
+        //Reserved 39
```

Comment:
Small reservation gap, will inevitably result in jumping past pmmp->discord packets...

Maybe time to consider shifting everything to leave a bigger gap before locking in network protocol.